cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -110,14 +113,25 @@ private void publishMetric(Iterable<TopicPayload> topicPayloads) throws IOExcept
     private final TopicId topicId;
     private final List<byte[]> payloads;
     private final RetryStrategy retryStrategy;
+    private int payloadSize;
+    private int metricsCount;
+    private Map<String, String> metricsTags;
+
 
     private TopicPayload(TopicId topicId, RetryStrategy retryStrategy) {
       this.topicId = topicId;
       this.retryStrategy = retryStrategy;
       this.payloads = new ArrayList<>();
+      this.payloadSize = 0;
+      this.metricsCount = 0;
     }
 
-    void addPayload(byte[] payload) {
+    void addPayload(byte[] payload, Map<String, String> metricsTags, int metricsCount) {
+      payloadSize += payload.length;
```

Comment:
is this, in the end, the size of the message?

Reply:
yes,  we add list of payloads (byte array) along with transactionWritePointer and construct a Generic record and pass in the body of the request in `ClientMessagingService`.

Reply:
updated the error message format

Reply:
no, the transaction write pointer is a long.

Reply:
ok, then it will not be that misleading.